core/src/main/java/org/mule/util/store/PersistentObjectStorePartition.java

```diff
@@ -265,9 +265,20 @@ private synchronized void loadStoredKeysAndFileNames() throws ObjectStoreExcepti
             File[] files = listValuesFiles();
             for (int i = 0; i < files.length; i++)
             {
-                File file = files[i];
-                StoreValue<T> storeValue = deserialize(file);
-                realKeyToUUIDIndex.put(storeValue.getKey(), file.getName());
+                try
+                {
+                    File file = files[i];
+                    StoreValue<T> storeValue = deserialize(file);
+                    realKeyToUUIDIndex.put(storeValue.getKey(), file.getName());
+                }
+                catch (ObjectStoreException e)
+                {
+                    if (logger.isWarnEnabled())
+                    {
+                        logger.warn("Could not deserialize an ObjectStore file");
```

Comment:
This should also make it clear that the data will be lost... Careful with the language you use not to freak the user out